### PR TITLE
NOTIF-216 Aligning payload (REST and Kafka)

### PR DESCRIPTION
**Pull request by theute**
PS: I don't think /notifications/sample is used in GwResouce.java

### CODE REVIEW

File: src/main/java/com/redhat/cloud/notifications/RestAction.java [hunk 1]
```diff
@@ -55,7 +55,12 @@
     @JsonProperty("account_id")
     public String accountId;
 
-    public java.util.Map<String,Object> payload;
+    @NotNull
+    public List<RestEvent> events;
+
+    @NotNull
+    public String context;
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/notifications/RestAction.java:62]: This maps 1-1 with what we have in the avro schema, but I think we can be less harsh with the user and use a `Map<String, Object>` instead (the same with the payload)

File: src/main/java/com/redhat/cloud/notifications/RestEvent.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+public class RestEvent {
+    @NotNull
+    @NotEmpty
+    public String payload;
+
+    @NotNull
+    public String metadata;
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/notifications/RestEvent.java:28]: This one should be an empty object: e.g. `RestMetadata` with no data.

**Review by josejulio — APPROVED** [review]
lgtm
There are some details can we can follow up, like decide if want to align 1-1 with the kafka  message or allow the user to pass the `context` and `payload` as json objects instead  of strings.

**Comment by theute** [comment]
good points... I might fix that before merging

**Pull request merged**